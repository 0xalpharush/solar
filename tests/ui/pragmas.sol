// Commented out are lexed as empty literals "0."
pragma solidity *.*.*;
pragma solidity *.*.0 ;
pragma solidity *.*.0;
pragma solidity *.*;
pragma solidity *.0 .*;
pragma solidity *.0 .0 ;
pragma solidity *.0 .0;
pragma solidity *.0 ;
// pragma solidity *.0.*;
pragma solidity *.0.0 ;
pragma solidity *.0.0;
pragma solidity *.0;
pragma solidity *;
pragma solidity 0 .*.*;
pragma solidity 0 .*.0 ;
pragma solidity 0 .*.0;
pragma solidity 0 .*;
pragma solidity 0 .0 .*;
pragma solidity 0 .0 .0 ;
pragma solidity 0 .0 .0;
pragma solidity 0 .0 ;
// pragma solidity 0 .0.*;
pragma solidity 0 .0.0 ;
pragma solidity 0 .0.0;
pragma solidity 0 .0;
pragma solidity 0 ;
// pragma solidity 0.*.*;
// pragma solidity 0.*.0 ;
// pragma solidity 0.*.0;
// pragma solidity 0.*;
pragma solidity 0.0 .*;
pragma solidity 0.0 .0 ;
pragma solidity 0.0 .0;
pragma solidity 0.0 ;
pragma solidity 0.0.*;
pragma solidity 0.0.0 ;
pragma solidity 0.0.0;
pragma solidity 0.0;
pragma solidity 0;

// pragma foo bar;
// ~^ ERROR unknown pragma
pragma abicoder v2;
// pragma solidity ^4294967295;
// ~^ ERROR too large
pragma solidity 0 - 1 0 - 2;
//~^ ERROR ranges can only be combined using the || operator
pragma abicoder "v2";
pragma solidity ^0.5.16 =0.8.22 || >=0.8.21 <=2 ~1 0.6.2;
pragma solidity 0.4 - 1 || 0.3 - 0.5.16;
