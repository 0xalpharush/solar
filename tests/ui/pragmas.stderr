error: ranges can only be combined using the || operator
  --> $DIR/pragmas.sol:47:1
   |
LL | pragma solidity 0 - 1 0 - 2;
   | ^^^^^^^^^^^^^^^^^^^^^
   |

error: aborting due to 1 previous error

